crates/sui-core/src/execution_engine.rs

```diff
@@ -729,3 +908,106 @@ fn test_pay_success_delete_all() {
     let owner_objs = store.get_written_objects_owned_by(&sender);
     assert!(owner_objs.is_empty());
 }
+
+#[test]
+fn test_pay_sui_success_one_input_coin() {
```

Comment:
here I wanted to test with non-Sui coins and verify that `NonSuiCoin` Error is returned, wondering is there always a non-sui token, for example right after genesis?